src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs

```diff
@@ -270,6 +272,20 @@ private IEnumerable<string> GetApplicationNames()
                 return appName.Split(StringSeparators.Comma);
             }
 
+            if (ApplicationNameIsBlankOrDefault())
```

Comment:
Do we even need this check? We could just say that the function name has a higher priority than the new relic config, it looks like we already do that for iisexpress and rolename. If customers want to override the function name default, they still have the option to do that via a .net configuration file or the new relic app name environment variable

Reply:
Yeah we could do it that way. It felt counterintuitive for a NuGet deploy, since the user wouldn't be able to set the name via the `newrelic.config` as they usually do. But maybe you're right, it might be closer to the IISExpress case.